src/mame/rm/rm380z_m.cpp

```diff
@@ -100,6 +134,13 @@ uint8_t rm380z_state::port_read(offs_t offset)
 
 	switch (offset)
 	{
+	case 0x00:
+		// bit 0 is low during HRG frame blanking
+		// bit 1 is low duing HRG line blanking
+		// (this is the inverse of VDU port 1 shifted 6 bits to the right)
+		data = ((m_port1 & 0xf0) >> 6) ^ 0x03;
+		break;
```

Comment:
Why are you masking to isolate four bits, then shifting two of them off the end?  It’s less error-prone to mask *after* shifting, or to use the `BIT` helpers, e.g. this could be expressed as `data = BIT(~m_port1, 6, 2);` (select two bits, starting at bit 6, counting from LSB).

That aside, this isn’t going to work properly – you aren’t setting the horizontal blanking flag anywhere.  You need to configure the screen device with proper timings (preferably using `set_raw` so everything is based on the pixel clock), and then use the `hblank()` and `vblank()` member functions of `screen_device` to get the current state.  Otherwise you’re liable to get out of sync with the screen device.  The whole `static_vblank_timer` is a dreadful hack to avoid actually configuring the screen device properly and it needs to go away.

Reply:
I don't need to mask at all so have changed this.  All I meant to do was take the two blanking bits from the main display and convert them to the HRG format (different positions and inverse values).

It does work in so far as they change often enough to allow the software to write to the screen.  The horizontal blanking flag is being set by rm380z_state::static_vblank_timer, but as you say it's a dreadful hack.  Nonetheless it allows software to recognise the card and update the display.  Software typically just waits for the blanking period as the CPU can only access video ram when it's not being used by the display (on real hardware).  For very quick updates it can use the horizontal blanking period, but for most things it waits for the vertical blank.  The timing does effect how fast software runs as it might be spending more or less time waiting than it should of course.  Things seem to run OK on the emulator, but I've never used real hardware so cannot say if it's running noticeably faster or slower than it should!